eduvpn/ui/ui.py

```diff
@@ -593,8 +604,61 @@ def on_switch_connection_state(self, switch, state):
                 self.app.interface_transition('activate_connection')
             else:
                 self.app.interface_transition('deactivate_connection')
+                self.stop_connection_info()
         return True
 
+    def stop_connection_info(self):
+        if self.connection_info_thread_cancel:
+            self.connection_info_thread_cancel()
+            self.connection_info_thread = None
+
+        if self.connection_info_stats:
+            self.connection_info_stats.cleanup()
+            self.connection_info_stats = None
+
+    def start_connection_info(self, check_expanded=True):
+        # Make sure to only update when we are expanded
+        if check_expanded:
+            is_expanded = self.connection_info_expander.get_expanded()
+            if not is_expanded:
+                return
+
+        if not self.app.network_state.has_transition('disconnect'):
```

Comment:
If you don't show based on network state here, will the stats show when the network state changes back? I think it's better to just show the stats regardless of network state, however.

Reply:
Do you mean that I should show the connection stats when a VPN is not active?

Reply:
You can reopen this if this is not right what I just pushed, but I think it does make sense as an extra error check